File: api/src/Entity/Animal.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'Animal')]
class Animal
{
    #[ORM\Id]
    #[ORM\ManyToOne(targetEntity: Espece::class, inversedBy: 'animaux')]
    #[ORM\JoinColumn(name: 'id_Espece', referencedColumnName: 'id_Espece', nullable: false)]
    private Espece $espece;

    #[ORM\Id]
    #[ORM\Column(name: 'nomBapteme_Animal', length: 50)]
    private string $nomBapteme;

    #[ORM\Column(name: 'dateNaissance_Animal', type: 'date', nullable: true)]
    private ?\DateTimeInterface $dateNaissance = null;

    #[ORM\Column(name: 'dateDeces_Animal', type: 'date', nullable: true)]
    private ?\DateTimeInterface $dateDeces = null;

    #[ORM\Column(name: 'genre_Animal', length: 50)]
    private string $genre;

    public function getEspece(): Espece { return $this->espece; }
    public function setEspece(Espece $espece): static { $this->espece = $espece; return $this; }
    public function getNomBapteme(): string { return $this->nomBapteme; }
    public function setNomBapteme(string $nomBapteme): static { $this->nomBapteme = $nomBapteme; return $this; }
    public function getDateNaissance(): ?\DateTimeInterface { return $this->dateNaissance; }
    public function setDateNaissance(?\DateTimeInterface $dateNaissance): static { $this->dateNaissance = $dateNaissance; return $this; }
    public function getDateDeces(): ?\DateTimeInterface { return $this->dateDeces; }
    public function setDateDeces(?\DateTimeInterface $dateDeces): static { $this->dateDeces = $dateDeces; return $this; }
    public function getGenre(): string { return $this->genre; }
    public function setGenre(string $genre): static { $this->genre = $genre; return $this; }
}

File: mobile/app/src/main/java/com/sigansud/app/models/Animal.java
package com.sigansud.app.models;

import com.google.gson.annotations.SerializedName;

public class Animal {

    @SerializedName("idEspece")
    private int idEspece;

    @SerializedName("nomBapteme")
    private String nomBapteme;

    @SerializedName("dateNaissance")
    private String dateNaissance;

    @SerializedName("dateDeces")
    private String dateDeces;

    @SerializedName("genre")
    private String genre;

    private Espece espece;

    public Animal() {}

    public int getIdEspece() { return idEspece; }
    public void setIdEspece(int idEspece) { this.idEspece = idEspece; }

    public String getNomBapteme() { return nomBapteme; }
    public void setNomBapteme(String nomBapteme) { this.nomBapteme = nomBapteme; }

    public String getNom() { return nomBapteme; }

    public String getDateNaissance() { return dateNaissance; }
    public void setDateNaissance(String dateNaissance) { this.dateNaissance = dateNaissance; }

    public String getDateDeces() { return dateDeces; }
    public void setDateDeces(String dateDeces) { this.dateDeces = dateDeces; }

    public String getGenre() { return genre; }
    public void setGenre(String genre) { this.genre = genre; }

    public Espece getEspece() { return espece; }
    public void setEspece(Espece espece) { this.espece = espece; }

    public boolean isDecede() {
        return dateDeces != null && !dateDeces.isEmpty();
    }
}
